src/bindings/jna/libelektra4j/src/main/java/org/libelektra/Key.java

```diff
@@ -333,120 +397,131 @@ public Key nextMeta() {
 	 * Getter for meta information
-	 * @param metaName Key name of meta information to be fetched
+	 *
+	 * @param metaName
+	 *            Key name of meta information to be fetched
 	 * @return New Key object containing the requested meta information
 	 */
-	public Key getMeta(String metaName) {
+	public Key getMeta(final String metaName) {
 		return new Key(Elektra.INSTANCE.keyGetMeta(get(), metaName));
 	}
 
 	/**
 	 * Setter for meta information
-	 * @param metaName Key name of meta information to be set
-	 * @param newMetaString Meta value to be set
-	 * @return -1 in case of an error, 0 if no meta with given name
-	 * 		is available for the key and value > 0 representing the
-	 * 		size of newMetaString if update successful
+	 *
+	 * @param metaName
+	 *            Key name of meta information to be set
+	 * @param newMetaString
+	 *            Meta value to be set
+	 * @return -1 in case of an error, 0 if no meta with given name is available for the key and value > 0 representing the size of
+	 *         newMetaString if update successful
 	 */
-	public int setMeta(String metaName, String newMetaString) {
-		return Elektra.INSTANCE.keySetMeta(get(), metaName,
-				newMetaString);
+	public int setMeta(final String metaName, final String newMetaString) {
+		return Elektra.INSTANCE.keySetMeta(get(), metaName, newMetaString);
 	}
 
 	/**
-	 * Helper function to compare two keys. Compares the key name
-	 * with normal String comparison.
-	 * @param other Other Key object that is used in comparison
-	 * @return 0 if key name is equal; -1 if this key name has lower
-	 * 		alphabetical order than the other key; 1 if this key has
-	 * 		higher alphabetical order
+	 * Helper function to compare two keys. Compares the key name with normal String comparison.
+	 *
+	 * @param other
+	 *            Other Key object that is used in comparison
+	 * @return 0 if key name is equal; -1 if this key name has lower alphabetical order than the other key; 1 if this key has higher
+	 *         alphabetical order
 	 */
-	public int cmp(Key other) {
+	public int cmp(final Key other) {
 		return Elektra.INSTANCE.keyCmp(get(), other.get());
 	}
 
 	/**
 	 * Helper function to check for relation between keys
-	 * @param other Other Key object that is used in relation check
-	 * @return 0 if other is equal to this; > 0 if other is sub-key of this key;
-	 * 		< 0 otherwise or in case of an error
+	 *
+	 * @param other
+	 *            Other Key object that is used in relation check
+	 * @return 0 if other is equal to this; > 0 if other is sub-key of this key; < 0 otherwise or in case of an error
 	 */
-	public int rel(Key other) {
+	public int rel(final Key other) {
 		return Elektra.INSTANCE.keyRel(get(), other.get());
 	}
 
 	/**
 	 * Helper function to check if synchronization is necessary
-	 * @return 1 if needs sync, 0 if no change done and -1 in case of
-	 * 		a null pointer
+	 *
+	 * @return 1 if needs sync, 0 if no change done and -1 in case of a null pointer
```

Comment:
I checked out doxygen and it supports javadocs basically, though not as nicely but good enough for now i think. It does parse the @return correctly, so i think theres no need to change that.

In my last commit i added the jna folder to the Doxyfile. But i had to set this EXTRACT_ALL option to true in addition, otherwise it seems not to extract the javadocs from the source files. However, this might has some bad implications on the other documentation, do you have any idea why it didn't worked without changing EXTRACT_ALL? I also set the optimize for c output to false, as it mentioned in the config file you should only set this to true if you only have c files, but we have c, c++, java, and there are also some python files now in my documentation (i suspect those are also extracted due to EXTRACT_ALL?)